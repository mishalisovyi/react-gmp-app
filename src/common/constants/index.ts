export const Icon = {
  VERTICAL_ELLIPSIS: String.fromCharCode(8942),
  MULTIPLICATION_X: String.fromCharCode(10005),
  TELEPHONE_RECORDER: String.fromCharCode(8981),
};

export const API_URL = 'http://localhost:4000';

export const HTTP_RESPONSE_STATUS_NO_CONTENT = 204;
