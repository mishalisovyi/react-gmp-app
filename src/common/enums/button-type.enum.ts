export enum ButtonType {
  Primary,
  Secondary,
  Tertiary,
  Menu,
  Transparent,
  Search,
}
