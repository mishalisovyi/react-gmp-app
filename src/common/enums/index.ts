export { ButtonType } from './button-type.enum';
export { HttpMethod } from './http-method.enum';
