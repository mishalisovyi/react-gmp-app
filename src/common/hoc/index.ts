export { withPortalRendering } from './withPortalRendering/withPortalRendering';
export { ModalWindow } from './ModalWindow/ModalWindow';
