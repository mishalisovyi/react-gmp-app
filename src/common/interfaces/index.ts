export { EnumerableComponentProps } from './EnumerableComponentProps';
export { ModalWindowProps, ModalWindowActionButtonProps } from './ModalWindowProps';
export { SelectListData, SelectListValue } from './SelectListData';
