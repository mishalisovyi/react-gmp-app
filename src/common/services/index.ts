export {
  sendGetRequest,
  sendPostRequest,
  sendPutRequest,
  sendDeleteRequest,
} from './http.service';
