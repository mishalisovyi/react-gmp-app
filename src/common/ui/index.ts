export { Button } from './Button/Button';
export { Divider } from './Divider/Divider';
export { DropdownMenu } from './DropdownMenu/DropdownMenu';
export { FilterSelectList } from './FilterSelectList/FilterSelectList';
export { LabelPrimary } from './LabelPrimary/LabelPrimary';
export { LabelSecondary } from './LabelSecondary/LabelSecondary';
export { Logo } from './Logo/Logo';
export { PopupConfirmation } from './PopupConfirmation/PopupConfirmation';
export { PopupDialog } from './PopupDialog/PopupDialog';
export { PopupInfo } from './PopupInfo/PopupInfo';
export { SearchPanel } from './SearchPanel/SearchPanel';
export { Spinner } from './Spinner/Spinner';
export { Tab } from './TabBar/Tab/Tab';
export { TabBar } from './TabBar/TabBar';
export { TextField } from './TextField/TextField';
