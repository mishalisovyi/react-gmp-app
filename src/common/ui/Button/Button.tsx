import React, { memo, MouseEventHandler, useMemo } from 'react';

import { ButtonType } from 'common/enums';

interface ButtonProps {
  type: ButtonType;
  text: string;
  onClick?: MouseEventHandler<HTMLButtonElement>;
}

const getCssClassesByButtonType = (type: ButtonType): string => {
  return {
    [ButtonType.Primary]: 'button--primary',
    [ButtonType.Secondary]: 'button--secondary',
    [ButtonType.Tertiary]: 'button--small button--tertiary',
    [ButtonType.Menu]: 'button--menu',
    [ButtonType.Transparent]: 'button--transparent',
    [ButtonType.Search]: 'button--search',
  }[type];
};

function ButtonComponent({ onClick, type, text }: ButtonProps) {
  const buttonClasses = useMemo(() => getCssClassesByButtonType(type), [type]);

  return <button type="button" className={`button ${buttonClasses}`} onClick={onClick}>{text}</button>;
}

ButtonComponent.defaultProps = {
  onClick: () => { },
};

export const Button = memo(ButtonComponent);
