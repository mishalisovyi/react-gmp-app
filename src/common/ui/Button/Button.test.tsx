import React from 'react';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { ButtonType } from 'common/enums';
import { Button } from './Button';

it('should render primary button', () => {
  const tree = renderer.create(<Button type={ButtonType.Primary} text="primary button" onClick={() => { }} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render secondary button', () => {
  const tree = renderer.create(<Button type={ButtonType.Secondary} text="secondary button" onClick={() => { }} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render tertiary button', () => {
  const tree = renderer.create(<Button type={ButtonType.Tertiary} text="tertiary button" onClick={() => { }} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render menu button', () => {
  const tree = renderer.create(<Button type={ButtonType.Menu} text="menu button" onClick={() => { }} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render transparent button', () => {
  const tree = renderer.create(<Button type={ButtonType.Transparent} text="transparent button" onClick={() => { }} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render search button', () => {
  const tree = renderer.create(<Button type={ButtonType.Search} text="search button" onClick={() => { }} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should handle click', async () => {
  const clickHandler = jest.fn();
  const buttonText = 'search button';

  render(<Button type={ButtonType.Search} text={buttonText} onClick={clickHandler} />);

  await userEvent.click(screen.getByText(buttonText));

  expect(clickHandler).toHaveBeenCalled();
});
