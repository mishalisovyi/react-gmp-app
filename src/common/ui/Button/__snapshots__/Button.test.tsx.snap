// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render menu button 1`] = `
<button
  className="button button--menu"
  onClick={[Function]}
  type="button"
>
  menu button
</button>
`;

exports[`should render primary button 1`] = `
<button
  className="button button--primary"
  onClick={[Function]}
  type="button"
>
  primary button
</button>
`;

exports[`should render search button 1`] = `
<button
  className="button button--search"
  onClick={[Function]}
  type="button"
>
  search button
</button>
`;

exports[`should render secondary button 1`] = `
<button
  className="button button--secondary"
  onClick={[Function]}
  type="button"
>
  secondary button
</button>
`;

exports[`should render tertiary button 1`] = `
<button
  className="button button--small button--tertiary"
  onClick={[Function]}
  type="button"
>
  tertiary button
</button>
`;

exports[`should render transparent button 1`] = `
<button
  className="button button--transparent"
  onClick={[Function]}
  type="button"
>
  transparent button
</button>
`;
